hack/scripts/prow-ci-operator-deploy.sh

```diff
@@ -1,4 +1,205 @@
 #!/bin/bash
 
-echo "Test Integration WIP"
-echo $NAMESPACE
+set -eo pipefail
+
+export IMAGE_NAME=aws-account-operator
+export BUILD_CONFIG=aws-account-operator
+export OPERATOR_DEPLOYMENT=aws-account-operator
+## Remove Below Line
+OC="oc"
+OC_WITH_NAMESPACE="$OC -n $NAMESPACE"
+
+function usage {
+    cat <<EOF
+    USAGE: $0 [ OPTION ]
+
+    OPTIONS:
+
+    --use-envrc     Uses the .envrc for setting up configmap and secret otherwise pick up from ci vault secrets
+    -n|--namespace  Sets the namespace to use
+    --is-local      Uses settings and makefile targets for local deployment
+    --is-staging    Uses settings and makefile targets for staging deployment
+EOF
+}
+
+function parseArgs {
+    PARSED_ARGUMENTS=$(getopt -o 'n:' --long 'namespace:,use-envrc,is-local,is-staging' -- "$@")
+    eval set -- "$PARSED_ARGUMENTS"
+
+    while :
+    do
+        case "$1" in
+            --use-envrc)
+                USE_ENVRC=1; shift
+                ;;
+            --is-local)
+                IS_LOCAL=1; shift
+                ;;
+            --is-staging)
+                IS_STAGING=1; shift
+                ;;
+            -n|--namespace)
+                NAMESPACE="$2";	shift 2
+                ;;
+            --)
+                shift
+                break
+                ;;
+            *)
+                echo "Unexpected option: $1."
+                usage
+                break
+                ;;
+        esac
+    done
+
+    echo "USE_ENVRC=${USE_ENVRC}"
+    echo "NAMESPACE=${NAMESPACE}"
+    echo "IS_LOCAL=${IS_LOCAL}"
+    echo "IS_STAGING=${IS_STAGING}"
+}
+
+function cleanup {
+    echo -e "\nPERFORMING CLEANUP\n"
+    ##  Add step to remove CRDs
+    if [[ $($OC get namespace $NAMESPACE --no-headers | wc -l) == 0 ]];
+        then
+            echo -e "\nno $NAMESPACE namespace found\n"
+            return
+	fi
+
+    $OC delete namespace $NAMESPACE || true
+}
+
+function cleanupPre {
+    cleanup    
+}
+
+function getEnvVariables {
+    if [[ -z $USE_ENVRC ]];
+        then
+            export FAKE_KEY_VALUE=$(cat /tmp/secret/aao-aws-creds/FAKE_KEY_VALUE)
```

Comment:
Where do all these files in `/tmp` come from - are they created somewhere?

Otherwise might be worth adding some documentation about this case.

Reply:
These files in `/tmp` are the secrets we created on vault UI, using following doc:-
https://docs.ci.openshift.org/docs/how-tos/adding-a-new-secret-to-ci/

Reply:
Added the above document link as a comment before using secret values.

Reply:
That's great - that makes it a lot clearer when and why those are used.